src/Features/LanguageServer/Protocol/Handler/References/FindUsagesLSPContext.cs

```diff
@@ -160,23 +160,23 @@ public override async ValueTask OnReferenceFoundAsync(SourceReferenceItem refere
                     _definitionsWithoutReference.Remove(definitionId);
                 }
 
+                // give this reference a fresh id.
                 _id++;
 
                 // Creating a new VSReferenceItem for the reference
                 var referenceItem = await GenerateVSReferenceItemAsync(
-                    definitionId, reference.SourceSpan,
+                    definitionId, _id, reference.SourceSpan,
                     reference.AdditionalProperties, definitionText: null,
                     definitionGlyph: Glyph.None, reference.SymbolUsageInfo, reference.IsWrittenTo, cancellationToken).ConfigureAwait(false);
 
                 if (referenceItem != null)
-                {
-                    _workQueue.AddWork(referenceItem);
-                }
+                    _workQueue.AddWork(referenceItem.Value);
             }
         }
 
-        private async Task<VSInternalReferenceItem?> GenerateVSReferenceItemAsync(
-            int? definitionId,
+        private async Task<SumType<VSInternalReferenceItem, LSP.Location>?> GenerateVSReferenceItemAsync(
+            int definitionId,
+            int id,
```

Comment:
`int?` was odd as this is always an int.  i moved to explicitly passing `id` down to so the helper could be decoupled from the state in this type.